app/models/manageiq/providers/ansible_tower/automation_manager/event_catcher/stream.rb

```diff
@@ -22,6 +23,7 @@ def poll
       catch(:stop_polling) do
         begin
           loop do
+            @before_poll.call if @before_poll && @before_poll.respond_to?(:call)
```

Comment:
I usually don't prefer `try` but this time I actually like `@before_poll.try(:call)` 

Note, `@before_poll` doesn't scream "proc/lambda" in my head.  I wonder if it makes sense to make this interface more clear by calling it `@before_poll_proc` or `@pre_poll_proc`